src/app/indexed-db/db-changes.service.ts

```diff
@@ -250,21 +278,16 @@ export class DbChangesService {
         }
         await this.contactIdbService.setContacts();
       }
-      //update NEBs
-      let nonEnergyBenefits: Array<IdbNonEnergyBenefit> = this.nonEnergyBenefitsIdbService.getCompanyNonEnergyBenefits(keyPerformanceIndicators[i].companyId);
-      if (nonEnergyBenefits.length > 0) {
-        for (let b = 0; b < nonEnergyBenefits.length; b++) {
-          let nonEnergyBenefit: IdbNonEnergyBenefit = nonEnergyBenefits[b];
-          let kpiMetricValues: Array<KeyPerformanceMetricValue> = keyPerformanceIndicators[i].performanceMetrics.flatMap(metric => {
-            return metric.value;
-          });
-          //remove metrics from this kpi
-          nonEnergyBenefit.performanceMetricImpacts = nonEnergyBenefit.performanceMetricImpacts.filter(pmi => {
-            return !kpiMetricValues.includes(pmi.kpmValue);
-          });
-          await firstValueFrom(this.nonEnergyBenefitsIdbService.updateWithObservable(nonEnergyBenefit));
+      //remove associated kpmImpacts
```

Comment:
kpmImpact is the impact of a NEB on a kpm, not sure if the kpmImpacts should be removed when a KPI is deleted. Need confirm with @koay9f

Reply:
The KPM that is being impacted gets deleted when the KPI it is associated with gets deleted. I have added proper messaging so the user knows what they are deleting.